.github/workflows/release-github-tag.yml

```diff
@@ -0,0 +1,54 @@
+name: Publish package to the Maven Central Repository
+
+on:
+  release:
+    types: [created]
+
+jobs:
+  release-github-tag:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Set up Maven Central Repository
+        uses: actions/setup-java@v1
+        with:
+          java-version: 1.8
+          server-id: ossrh
+          server-username: MAVEN_USERNAME
+          server-password: MAVEN_PASSWORD
+
+      - name: Build a project & run unit tests
+        run: mvn -B -ntp package
+
+      - name: Get the tag name
+        run: echo "TAG=${GITHUB_REF/refs\/tags\//}" >> $GITHUB_ENV
+
+      - name: Set version from git tag
+        run: mvn -B -ntp versions:set -DgenerateBackupPoms=false -DnewVersion="$TAG"
```

Comment:
You can avoid this step by using a revision.
See https://maven.apache.org/maven-ci-friendly.html